.github/workflows/cd-infrastructure.yml

```diff
@@ -27,4 +28,32 @@ jobs:
       azure_ad_client_secret: ${{ secrets.AZURE_AD_CLIENT_SECRET }}
       azure_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
       azure_ad_tenant_id: ${{ secrets.AZURE_AD_TENANT_ID }}
-      azure_storage_account_key: ${{ secrets.AZURE_STORAGE_ACCOUNT_KEY }}
\ No newline at end of file
+      azure_storage_account_key: ${{ secrets.AZURE_STORAGE_ACCOUNT_KEY }}
+  Dev_deployment:
+    if: github.event.pull_request.merged == true
```

Comment:
I did have this debate internally, the flow can be run with either as the workflow will also be tiggered by push to main - my only qualm would be in the case of needing to push a new workflow to main (in order to further develop it on a feature branch as we ran into the other week), and for that change to run/impact the dev/prod infrastructure without being fully developed. 

Let me know your thoughts on this as it is a quick change to switch to running from the push to main trigger :)

Reply:
@decodingahmed Correct me if I am wrong, but on line 4 I think we are referencing a push to main, and this is just ensuring that a PR request does not initiate the dev deploy since we have to initiation types pr and push?

Reply:
@adelechurchill please correct me if I misunderstood but I think you're saying you're concerned that we would trigger dev/prod deployments when other new workflows are merged into main?  I think we might be able to address this with moving workflows into relevant directories. See reply to yahya's comment above.

Something to consider might be to split this file into separate workflows:
- pr_workflow.yaml -- triggers on PR changes and manually only
- release_workflow.yaml -- triggers on commits into main and manually

That might reduce the need for adding these logical gates and workflows would become clearer. What do you think?

---

> @decodingahmed Correct me if I am wrong, but on line 4 I think we are referencing a push to main, and this is just ensuring that a PR request does not initiate the dev deploy since we have two initiation types pr and push?

@yahya130 yes that's correct

Reply:
@yahya130 @decodingahmed following on from todays parking lot discussions, this file has been split into two workflows.
- Release workflow - Development deployment occurs only on push to main, and Production provisioning occurs only after successful execution of Development deployment
- PR workflow - PR provisioning occurs only when a pull request is opened/synced (it will not run if a PR is closed/merged)